server/webdriver_tests/place_explorer_test.py

```diff
@@ -37,6 +40,43 @@ def test_page_serve(self):
             self.assertEqual(response.getcode(), 200)
         self.assertEqual("United States | Place Explorer | Data Commons", self.driver.title)
 
+    def test_place_search(self):
+        """Test the place search box can work correctly."""
+        self.driver.get(self.url_ + USA_URL)
+        # Using implicit wait here to wait for loading page.
+        self.driver.implicitly_wait(5)
+        search_border = self.driver.find_element_by_class_name("search")
+        search_box = search_border.find_element_by_class_name("pac-target-input")
+        search_box.send_keys(PLACE_SEARCH)
+        self.driver.implicitly_wait(3)
+        search_results = self.driver.find_elements_by_class_name("pac-item")
+        ca_result = search_results[0]
+        ca_result.click()
+        self.driver.implicitly_wait(3)
+        self.assertEqual("California | Place Explorer | Data Commons", self.driver.title)
+
+    def test_demograpgics_link_and_switch_to_similar(self):
+        """
+        Test the demographics link can work correctly, and when changing from containing to similar,
+        the chart should change accordingly.
+        """
+        self.driver.get(self.url_ + USA_URL)
+        time.sleep(5)
+        demographics = self.driver.find_element_by_id("Demographics")
+        demographics.find_element_by_tag_name('a').click()
+        time.sleep(5)
+        self.assertTrue("Demographics" in self.driver.current_url)
+        subtopics = self.driver.find_elements_by_class_name("subtopic")
+        age_topic = subtopics[1]
+        age_charts = age_topic.find_elements_by_class_name("col")
+        age_across_places_chart = age_charts[2]
+        origin = age_across_places_chart.find_element_by_class_name("svg-container").get_attribute("innerHTML")
+        selects = Select(age_across_places_chart.find_element_by_tag_name("select"))
+        selects.select_by_value("SIMILAR")
+        time.sleep(3)
+        after = age_across_places_chart.find_element_by_class_name("svg-container").get_attribute("innerHTML")
+        self.assertTrue(origin != after)
```

Comment:
This might not be a good one since the current USA page, similar place does not quite have data.

Can you use California instead, and assert that, inside svg, there are 3 <g> element, and the 3rd <g> element has >2 <g> elements (each corresponding to a group bar chart). You can make this a function and reuse across tests.

Reply:
Done

Reply:
I am not using this as a function, cause this is a little specific, and the code is short.